packages/labextension/src/connection.ts

```diff
@@ -78,9 +78,13 @@ export class NeptuneConnection {
   };
 
   getNotebook = () => {
+
     return this
       .getAuthorizationHeader()
-      .then(authorizationHeader => {
+      .then((authorizationHeader) => {
+        if (!this.params.notebookId) {
+          return {} as INeptuneNotebook;
```

Comment:
Frankly I think that returning empty object and saying to TypeScript that it is INeptuneNotebook (it does not have any optional field!) is a hack. However I see that other parts of the code are written in the same manner and fixing this small bug in a proper way would force you to quite big refactor of whole extension.

I think we can agree to merge as is as this is bug fix.
However when will add new features to the extension, we should consider refactoring.

Reply:
Agreed